mmaction/models/losses/hvu_loss.py

```diff
@@ -0,0 +1,131 @@
+            as negative samples.
+        loss_weight (float): The loss weight. Default: 1.0.
+    """
+
+    def __init__(self,
+                 categories=[
+                     'action', 'attribute', 'concept', 'event', 'object',
+                     'scene'
+                 ],
+                 category_nums=(739, 117, 291, 69, 1679, 248),
+                 category_loss_weights=(1, 1, 1, 1, 1, 1),
+                 loss_type='all',
+                 with_mask=False,
+                 loss_weight=1.0):
+
+        super().__init__(loss_weight)
+        self.categories = categories
+        self.category_nums = category_nums
+        self.category_loss_weights = category_loss_weights
+        assert len(self.category_nums) == len(self.category_loss_weights)
+        for loss_weight in self.category_loss_weights:
+            assert loss_weight >= 0
+        self.loss_type = loss_type
+        self.with_mask = with_mask
+        self.category_startidx = [0]
+        for i in range(len(self.category_nums) - 1):
+            self.category_startidx.append(self.category_startidx[-1] +
+                                          self.category_nums[i])
+        assert self.loss_type in ['individual', 'all']
+
+    def _forward(self, cls_score, label, aux_info):
+        """Forward function.
+
+        Args:
+            cls_score (torch.Tensor): The class score.
+            label (torch.Tensor): The ground truth label.
+            aux_info (dict[torch.Tensor]): Other auxiliary tensors needed for
+                loss calculation. For HVU Loss, `aux_info` should contain
+                `mask` and `category_mask`
+
+        Returns:
+            torch.Tensor: The returned CrossEntropy loss.
+        """
+        assert 'mask' in aux_info
+        assert 'category_mask' in aux_info
+        mask = aux_info['mask']
+        category_mask = aux_info['category_mask']
+
+        if self.loss_type == 'all':
+            loss_cls = F.binary_cross_entropy_with_logits(
+                cls_score, label, reduction='none')
+            if self.with_mask:
+                w_loss_cls = mask * loss_cls
+                w_loss_cls = torch.sum(
+                    w_loss_cls, dim=1) / torch.sum(
+                        mask, dim=1)
+                w_loss_cls = torch.mean(w_loss_cls)
+                return dict(loss_cls=w_loss_cls)
+            else:
+                return dict(loss_cls=torch.mean(loss_cls))
+        elif self.loss_type == 'individual':
+            losses = {}
+            loss_weights = {}
+            for name, num, start_idx in zip(self.categories,
+                                            self.category_nums,
+                                            self.category_startidx):
+                category_score = cls_score[:, start_idx:start_idx + num]
+                category_label = label[:, start_idx:start_idx + num]
+                category_loss = F.binary_cross_entropy_with_logits(
```

Comment:
https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/losses/cross_entropy_loss.py#L28

Reply:
Same as the last comment